modvendor/modvendor.go

```diff
@@ -148,3 +303,81 @@ func Vendor(rootdir string, modsrc tf.Source) (string, error) {
 	}
 	return modVendorDir, nil
 }
+
+func patchFiles(rootdir string, files []string, sourcemap map[string]*modinfo) error {
+	logger := log.With().
+		Str("action", "modvendor.patchFiles").
+		Logger()
+
+	logger.Trace().Msg("patching vendored files")
+
+	errs := errors.L()
+	for _, fname := range files {
+		logger := logger.With().
+			Str("filename", fname).
+			Logger()
+
+		bytes, err := ioutil.ReadFile(fname)
+		if err != nil {
+			errs.Append(err)
+			continue
+		}
+		parsedFile, diags := hclwrite.ParseConfig(bytes, fname, hcl.Pos{})
+		if diags.HasErrors() {
+			errs.Append(errors.E(diags))
+			continue
+		}
+
+		logger.Trace().Msg("successfully parsed for patching")
+
+		blocks := parsedFile.Body().Blocks()
+		for _, block := range blocks {
+			if block.Type() != "module" {
+				continue
+			}
+			if len(block.Labels()) != 1 {
+				continue
+			}
+			source := block.Body().GetAttribute("source")
+			if source == nil {
+				continue
+			}
+
+			sourceString := string(source.Expr().BuildTokens(nil).Bytes())
+			sourceString = strings.TrimSpace(sourceString)
+			sourceString = sourceString[1 : len(sourceString)-1] // unquote
```

Comment:
if tf code has a source = "" won't that parse OK but then panic here ?

Reply:
`source.Expr().BuildTokens(nil).Bytes()` will return `""` and then this line will return an empty string.
But I will add a test for this. I didn't have time to improve that yet. I think I'm gonna do it in a separate PR, I think we can avoid messing with tokens here.

Reply:
Hmm Im probably missing something stupid, if the sourceString is "" wont the slice be `sourceString[1:-1]` which will panic ?

Like this : https://go.dev/play/p/fl6CqsY2qTq

Again, probably missing something here :laughing:

Reply:
Ah now I got it, you mean the buildtokens never returns an actual empty string, it has always at least the double quotes. Makes sense.

Reply:
yep :smile:

Reply:
I meant it returns `OQUOTE` `STR` `CQUOTE`.